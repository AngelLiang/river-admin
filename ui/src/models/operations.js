export class CreateOrUpdateOperation{
    constructor(object) {
        this.operation_object = object
    }
}

export class DeleteOperation {
    constructor(objectId) {
        this.object_id = objectId
    }
}
