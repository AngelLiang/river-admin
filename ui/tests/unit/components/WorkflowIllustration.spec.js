import Vuetify from 'vuetify'
import Vue from 'vue'
import { mount } from '@vue/test-utils'
import { Workflow, State, Transition } from "../../../src/models/models"
import WorkflowIllustration from '@/components/WorkflowIllustration.vue'

describe('WorkflowIllustration.vue', () => {
    let vuetify

    beforeEach(() => {
        vuetify = new Vuetify()
        Vue.use(Vuetify)

        window.SVGElement.prototype.getBBox = () => ({
            x: Math.floor(Math.random() * Math.floor(100)),
            y: Math.floor(Math.random() * Math.floor(100)),
        });

    })

    afterEach(() => {
        delete window.SVGElement.prototype.getBBox;
    });

    it('should render even with one state', async () => {
        var state_1 = State.of(1, "state-1")
        var states = [state_1]
        const wrapper = mount_component({ states, transitions: [], editable: true, state_class_mapping: {} })
        expect(wrapper.find(`svg g#state_${state_1.id}`).exists()).toBeTruthy()
        expect(wrapper.find(`svg g#state_${state_1.id}`).classes()).toContainEqual("node-default")
        expect(wrapper.find(`svg g#state_${state_1.id} > rect`).exists()).toBeTruthy()
        expect(wrapper.find(`svg g#state_${state_1.id} > g.label tspan`).exists()).toBeTruthy()
        expect(wrapper.find(`svg g#state_${state_1.id} > g.label tspan`).text()).toBe(state_1.label)

        expect(wrapper.element).toMatchSnapshot()
    })

    it('should render with more than one state', async () => {
        var state_1 = State.of(1, "state-1")
        var state_2 = State.of(2, "state-2")
        var state_3 = State.of(3, "state-3")
        var states = [state_1, state_2, state_3]
        const wrapper = mount_component({ states, transitions: [], editable: true, state_class_mapping: {} })
        expect(wrapper.find(`svg g#state_${state_1.id}`).exists()).toBeTruthy()
        expect(wrapper.find(`svg g#state_${state_2.id}`).exists()).toBeTruthy()
        expect(wrapper.find(`svg g#state_${state_3.id}`).exists()).toBeTruthy()

        expect(wrapper.element).toMatchSnapshot()
    })


    it.skip('(library doesnt work in test)should render with one transition', async () => {
        var state_1 = State.of("1", "state-1")
        var state_2 = State.of("2", "state-2")
        var states = [state_1, state_2]

        var content_type = { "app_label": "test-app", "model": "test-model" }
        var workflow = Workflow.of("workflow-1", content_type, state_1, "test_field");
        var transition_1 = Transition.of("1", workflow, state_1.id, state_2.id)
        var transitions = [transition_1]

        const wrapper = mount_component({ states, transitions, editable: true, state_class_mapping: {} })
        expect(wrapper.find(`svg g#state_${state_1.id}`).exists()).toBeTruthy()
        expect(wrapper.find(`svg g#state_${state_2.id}`).exists()).toBeTruthy()

        expect(wrapper.find(`svg g.edgePaths > g.edgePath > path.path`).exists()).toBeTruthy()

        expect(wrapper.element).toMatchSnapshot()
    })


    function mount_component(propsData) {
        return mount(WorkflowIllustration, {
            vuetify,
            sync: false,
            propsData,
        })
    }
})