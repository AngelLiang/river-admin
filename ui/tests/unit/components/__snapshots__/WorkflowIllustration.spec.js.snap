// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowIllustration.vue should render even with one state 1`] = `
<div
  class="mx-auto pa-5 v-card v-sheet theme--light elevation-6"
>
  <div
    class="v-card__text"
  >
    <div
      class="container container--fluid"
    >
      <div
        class="row align-center justify-center"
      >
        <div
          align="center"
          class="col"
          id="svg-container"
          justify="center"
        >
          <svg
            class="shadow"
            height="NaN"
            width="650"
          >
            <g
              transform="translate(NaN, 20)"
            >
              <g
                class="output"
              >
                <g
                  class="clusters"
                />
                <g
                  class="edgePaths"
                />
                <g
                  class="edgeLabels"
                />
                <g
                  class="nodes"
                >
                  <g
                    class="node node-default"
                    id="state_1"
                    style="opacity: 1;"
                    transform="translate(NaN,NaN)"
                  >
                    <rect
                      class="label-container"
                      height="NaN"
                      rx="5"
                      ry="5"
                      width="NaN"
                      x="NaN"
                      y="NaN"
                    />
                    <g
                      class="label"
                      transform="translate(0,0)"
                    >
                      <g
                        transform="translate(NaN,NaN)"
                      >
                        <text>
                          <tspan
                            dy="1em"
                            space="preserve"
                            x="1"
                          >
                            state-1
                          </tspan>
                        </text>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkflowIllustration.vue should render with more than one state 1`] = `
<div
  class="mx-auto pa-5 v-card v-sheet theme--light elevation-6"
>
  <div
    class="v-card__text"
  >
    <div
      class="container container--fluid"
    >
      <div
        class="row align-center justify-center"
      >
        <div
          align="center"
          class="col"
          id="svg-container"
          justify="center"
        >
          <svg
            class="shadow"
            height="NaN"
            width="650"
          >
            <g
              transform="translate(NaN, 20)"
            >
              <g
                class="output"
              >
                <g
                  class="clusters"
                />
                <g
                  class="edgePaths"
                />
                <g
                  class="edgeLabels"
                />
                <g
                  class="nodes"
                >
                  <g
                    class="node node-default"
                    id="state_1"
                    style="opacity: 1;"
                    transform="translate(NaN,NaN)"
                  >
                    <rect
                      class="label-container"
                      height="NaN"
                      rx="5"
                      ry="5"
                      width="NaN"
                      x="NaN"
                      y="NaN"
                    />
                    <g
                      class="label"
                      transform="translate(0,0)"
                    >
                      <g
                        transform="translate(NaN,NaN)"
                      >
                        <text>
                          <tspan
                            dy="1em"
                            space="preserve"
                            x="1"
                          >
                            state-1
                          </tspan>
                        </text>
                      </g>
                    </g>
                  </g>
                  <g
                    class="node node-default"
                    id="state_2"
                    style="opacity: 1;"
                    transform="translate(NaN,NaN)"
                  >
                    <rect
                      class="label-container"
                      height="NaN"
                      rx="5"
                      ry="5"
                      width="NaN"
                      x="NaN"
                      y="NaN"
                    />
                    <g
                      class="label"
                      transform="translate(0,0)"
                    >
                      <g
                        transform="translate(NaN,NaN)"
                      >
                        <text>
                          <tspan
                            dy="1em"
                            space="preserve"
                            x="1"
                          >
                            state-2
                          </tspan>
                        </text>
                      </g>
                    </g>
                  </g>
                  <g
                    class="node node-default"
                    id="state_3"
                    style="opacity: 1;"
                    transform="translate(NaN,NaN)"
                  >
                    <rect
                      class="label-container"
                      height="NaN"
                      rx="5"
                      ry="5"
                      width="NaN"
                      x="NaN"
                      y="NaN"
                    />
                    <g
                      class="label"
                      transform="translate(0,0)"
                    >
                      <g
                        transform="translate(NaN,NaN)"
                      >
                        <text>
                          <tspan
                            dy="1em"
                            space="preserve"
                            x="1"
                          >
                            state-3
                          </tspan>
                        </text>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
